src/Orleans.Core/Streams/Internal/StreamImpl.cs

```diff
@@ -160,7 +160,9 @@ internal IInternalAsyncObservable<T> GetConsumerInterface()
 
         private IInternalStreamProvider GetStreamProvider()
         {
-            return this.runtimeClient.CurrentStreamProviderManager.GetProvider(streamId.ProviderName) as IInternalStreamProvider;
+            var streamProvider = this.runtimeClient.ServiceProvider.GetServiceByName<IStreamProvider>(streamId.ProviderName)
+                ?? throw new KeyNotFoundException(streamId.ProviderName);
+            return streamProvider as IInternalStreamProvider;
```

Comment:
(out of scope) this down-cast seems to limit the ability for users to implement `IStreamProvider` on their own, without using one our ordained base classes like `PersistentStreamProvider`

Reply:
This logic is shared code between our sms and persistent stream providers.  It's not associated with an extensibility point.  I agree that this is not 'good', but I doesn't really limit extensibility as the supported extensibility points are at the provider level or adapter level.